docker-build-proto/Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM golang:1.14-alpine3.11 as builder
+
+RUN apk add --update --no-cache build-base curl git upx && \
+  rm -rf /var/cache/apk/*
+
+ENV GOLANG_PROTOBUF_VERSION=1.3.1 \
+  GOGO_PROTOBUF_VERSION=1.3.1
+RUN GO111MODULE=on go get \
+  github.com/golang/protobuf/protoc-gen-go@v${GOLANG_PROTOBUF_VERSION} \
+  github.com/gogo/protobuf/protoc-gen-gogo@v${GOGO_PROTOBUF_VERSION} && \
+  mv /go/bin/protoc-gen-go* /usr/local/bin/
+
+ENV PROTOBUF_VERSION=3.11.4
+RUN mkdir -p /tmp/protoc && \
+  curl -sSL \
+  https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip \
+  -o /tmp/protoc/protoc.zip && \
+  cd /tmp/protoc && \
+  unzip protoc.zip && \
+  mv /tmp/protoc/include /usr/local/include
+
+RUN upx --lzma /usr/local/bin/*
+
+FROM alpine:edge
+
+WORKDIR /work
+
+ENV \
+  GRPC_VERSION=1.27.3 \
+  PROTOBUF_VERSION=3.11.4 \
+  ALPINE_GRPC_VERSION_SUFFIX=r0 \
+  ALPINE_PROTOBUF_VERSION_SUFFIX=r1
+
+RUN echo 'http://dl-cdn.alpinelinux.org/alpine/edge/testing' >> /etc/apk/repositories && \
+  apk add --update --no-cache bash curl git grpc=${GRPC_VERSION}-${ALPINE_GRPC_VERSION_SUFFIX} protobuf=${PROTOBUF_VERSION}-${ALPINE_PROTOBUF_VERSION_SUFFIX} && \
```

Comment:
Are these really necessary in the final image?

Reply:
removed bask curl and git as they are not needed, grpc and protobuf is needed